docs(service-worker): various small fixes/improvements in SW docs

gkalpak opened the pull request:
- _Commit 1:_ **update default value of `SwRegistrationOptions#registrationStrategy`**
  The default value was changed from `registerWhenStable` to `registerWhenStable:30000` in 29e8a64, but the decumentation was not updated to reflect that.

  This commit updates the documentation to mention the correct default value.

- _Commit 2:_ **docs(service-worker): minor fixes/improvements in the SW Communication guide**
  This commit includes various fixes/improvements for the "Service worker communication" guide.

##
This PR partially addresses #37527.

mary-poppins commented:
You can preview 01ca0d5 at https://pr37555-01ca0d5.ngbuilds.io/.

mary-poppins commented:
You can preview 9c78503 at https://pr37555-9c78503.ngbuilds.io/.

kapunahelewong commented on aio/content/guide/service-worker-communications.md:
```diff
@@ -31,28 +31,27 @@ You can use these events to notify the user of a pending update or to refresh th
 
-It's possible to ask the service worker to check if any updates have been deployed to the server. You might choose to do this if you have a site that changes frequently or want updates to happen on a schedule.
+It's possible to ask the service worker to check if any updates have been deployed to the server.
+The service worker checks for updates during initialization and on each navigation request (i.e. when the user navigates from a different address to your app).
```
```suggestion
The service worker checks for updates during initialization and on each navigation request&mdash;that is, when the user navigates from a different address to your app.
```
Or any other way you'd like to say it without parentheses or "e.g." (no latin).

kapunahelewong commented on aio/content/guide/service-worker-communications.md:
```diff
@@ -31,28 +31,27 @@ You can use these events to notify the user of a pending update or to refresh th
-
-You can avoid that by waiting for the app to stabilize first, before starting to poll for updates
-(as shown in the example above).
+In order to avoid negatively affecting the initial rendering of the page, `ServiceWorkerModule` will by default wait for up to 30 seconds for the app to stabilize, before registering the ServiceWorker script.
```
```suggestion
In order to avoid negatively affecting the initial rendering of the page, `ServiceWorkerModule` waits for up to 30 seconds by default for the app to stabilize, before registering the ServiceWorker script.
```

kapunahelewong commented on aio/content/guide/service-worker-communications.md:
```diff
@@ -31,28 +31,27 @@ You can use these events to notify the user of a pending update or to refresh th
-You can avoid that by waiting for the app to stabilize first, before starting to poll for updates
-(as shown in the example above).
+In order to avoid negatively affecting the initial rendering of the page, `ServiceWorkerModule` will by default wait for up to 30 seconds for the app to stabilize, before registering the ServiceWorker script.
+Constantly polling for updates, e.g. with [setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) or RxJS' [interval()](https://rxjs.dev/api/index/function/interval), will prevent the app from stabilizing and the ServiceWorker script will not be registered with the browser until the 30 seconds upper limit is reached.
```
```suggestion
Constantly polling for updates, for example, with [setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) or RxJS' [interval()](https://rxjs.dev/api/index/function/interval), will prevent the app from stabilizing and the ServiceWorker script will not be registered with the browser until the 30 seconds upper limit is reached.
```

kapunahelewong commented on aio/content/guide/service-worker-communications.md:
```diff
@@ -31,28 +31,27 @@ You can use these events to notify the user of a pending update or to refresh th
 Note that this is true for any kind of polling done by your application.
 Check the {@link ApplicationRef#isStable isStable} documentation for more information.
 
+You can avoid that delay by waiting for the app to stabilize first, before starting to poll for updates (as shown in the example above).
```
```suggestion
You can avoid that delay by waiting for the app to stabilize first, before starting to poll for updates , as in the example above.
```

kapunahelewong reviewed (approved):
Just a few style suggestions. Thank you, @gkalpak!

Reviewed-for: global-docs-approvers

gkalpak committed:
docs(service-worker): update default value of `SwRegistrationOptions#registrationStrategy`

The default value was changed from `registerWhenStable` to
`registerWhenStable:30000` in 29e8a64cf02fa9b80d38c30091e0b730403c54c8,
but the decumentation was not updated to reflect that.

This commit updates the documentation to mention the correct default
value.

gkalpak committed:
docs(service-worker): minor fixes/improvements in the SW Communication guide

This commit includes various fixes/improvements for the
"Service worker communication" guide.

This partially addresses #37527.

gkalpak committed:
fixup! docs(service-worker): minor fixes/improvements in the SW Communication guide

mary-poppins commented:
You can preview 96c8ec7 at https://pr37555-96c8ec7.ngbuilds.io/.
